src/us/kbase/workspace/test/database/mongo/MongoWorkspaceDBTest.java

```diff
@@ -345,6 +329,104 @@ private void checkWorkspaceUpdateTimes(final Set<Instant> wsTimes) {
 		assertThat("incorrect workspace count", gotTimes.size(), is(wsTimes.size()));
 	}
 	
+	private Set<ObjectIDResolvedWS> setupTestDataForHideDelete(
+			final MongoWorkspaceDB db,
+			final ResolvedWorkspaceID wsid,
+			final ResolvedWorkspaceID wsid2)
+			throws Exception {
+		// create workspaces
+		final WorkspaceUser u = new WorkspaceUser("u");
+		db.createWorkspace(u, "ws1", false, null, new WorkspaceUserMetadata());
+		db.createWorkspace(u, "ws2", false, null, new WorkspaceUserMetadata());
+		
+		// save objects
+		final Provenance p = new Provenance(u);
+		final String type = "Mod.Type-5.1";
+		
+		saveTestObject(db, wsid, u, p, "newobj", type, "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", 22L);
+		saveTestObject(db, wsid, u, p, "newobj2", type, "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab", 22L);
+		saveTestObject(db, wsid2, u, p, "newobj3", type, "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaac", 22L);
+		
+		final ObjectIDResolvedWS oid1 = new ObjectIDResolvedWS(wsid, "newobj");
+		final ObjectIDResolvedWS oid2 = new ObjectIDResolvedWS(wsid, "newobj2");
+		final ObjectIDResolvedWS oid3 = new ObjectIDResolvedWS(wsid2, "newobj3");
+		final Set<ObjectIDResolvedWS> objectIDs = set(oid1, oid2, oid3);
+		return objectIDs;
+	}
+	
+	@Test
+	public void hideAndUnhideObjects() throws Exception {
+		// a basic happy path test for un/hiding objects
+		// doesn't make much sense to split into a test for hide and a test for unhide since
+		// the setup for unhide is basically the hide test
+		// TODO TEST add deeper un/hide unit tests, including unhappy path. Scan code paths
+		// Most tests are in WorkspaceTest.java which is an integration test
+		
+		// setup mocks
+		final TestMocks mocks = getDBInstance();
+		final MongoWorkspaceDB db = mocks.mdb;
+		when(mocks.clockmock.instant()).thenReturn(
+				inst(10000), // save #1 ws update
+				inst(10000), // save #2 ws update
+				inst(10000), // save #3 ws update
+				inst(20000), // hide objects in one ws
+				inst(40000), // hide objects in other ws
+				inst(60000), // unhide objects in one ws
+				inst(80000)  // unhide objects in other ws
+				);
+		
+		final ResolvedWorkspaceID wsid = new ResolvedWorkspaceID(1, "ws", false, false);
+		final ResolvedWorkspaceID wsid2 = new ResolvedWorkspaceID(2, "ws2", false, false);
+
+		final Set<ObjectIDResolvedWS> objectIDs = setupTestDataForHideDelete(db, wsid, wsid2);
+
+		hideOrUnhideAndCheck(db, objectIDs, true, wsid, wsid2, set(inst(20000), inst(40000)));
+		hideOrUnhideAndCheck(db, objectIDs, false, wsid, wsid2, set(inst(60000), inst(80000)));
+	}
+	
+	private void hideOrUnhideAndCheck(
```

Comment:
I tried combining this method and the equivalent method for delete, but it became an unreadable mess with a zillion method arguments (and there's already a lot)